plugins/PagePerformance/Controller.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\PagePerformance;
+
+use Piwik\API\Request;
+use Piwik\Common;
+use Piwik\DataTable;
+use Piwik\Metrics\Formatter;
+use Piwik\Piwik;
+use Piwik\Plugin\Controller as PluginController;
+use Piwik\Plugin\ReportsProvider;
+use Piwik\Plugins\CoreVisualizations\Visualizations\JqplotGraph\Evolution as EvolutionViz;
+use Piwik\Plugins\PagePerformance\Visualizations\JqplotGraph\StackedBarEvolution;
+use Piwik\View;
+use Piwik\ViewDataTable\Factory as ViewDataTableFactory;
+
+class Controller extends PluginController
+{
+    public function indexPagePerformance()
+    {
+        $view = new View('@PagePerformance/getPagePerformancePopover');
```

Comment:
I suppose permissions are checked in the API method. Might still be good to call `$this->checkSitePermission();` just like in the other methods.